### PR TITLE
[FCE-240] Add android-sdk to repo

### PR DESCRIPTION
**Pull request by karkakol**
## Description

Goal of this PR is to add android native repo.

## Motivation and Context

To make working with our project much easier

## How has this been tested?

I installed app from jitpack, run e2e testst locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Things to consider:
- [ ] change name of this repo (eg fishjam-cloud/mobile-clients)
- [ ] change package name of android client 
- [ ] fix naming in repo
- [ ] remove sdk postfixes everywhere
- [ ] react native android gradle should have the newest released jitpack version
- [ ] right now working on this project would always use android local version, it should be considered to use the newest jitpack version somewhere in testing
- [ ] remove android app?
- [ ] unify packages versions and releases ( how to deploy them all in the smart way)

### CODE REVIEW

File: .github/workflows/react_native_release.yaml
Comment (graszka22) [inline .github/workflows/react_native_release.yaml:2]: how do we release android package?
do we release all packages (rn and android) at once with the same version or separately?

File: compile_proto.sh
Comment (graszka22) [inline compile_proto.sh:13]: we don't need this for RN

File: examples/video-chat/android/settings.gradle
Comment (graszka22) [inline examples/video-chat/android/settings.gradle:8]: consider more verbose name (for example that contains fishjam)
if a library user has a module named "android-client" the build will break and "android-client" doesn't tell much

File: packages/android-client/app/build.gradle
Comment (graszka22) [inline packages/android-client/app/build.gradle:1]: do we want android example? if so it should go to examples folder, right?
but it seems we don't maintain the android example anyway

File: .github/workflows/android_build_lint_test.yaml
Comment (graszka22) [inline .github/workflows/android_build_lint_test.yaml:1]: can we add a check to run this only if we modify anything in packages/android-client folder?

File: packages/android-client/.release-it.json
Comment (graszka22) [inline packages/android-client/.release-it.json:1]: do we still use release-it for android releases?

File: packages/react-native-client/android/build.gradle
Comment (graszka22) [inline packages/react-native-client/android/build.gradle:85]: so do we need to update this manually on every release?

File: packages/android-client/FishjamClient/build.gradle
Comment (graszka22) [inline packages/android-client/FishjamClient/build.gradle:16]: this was copied from the upper-level gradle file, can we avoid this somehow?

also in the upper gradle file there is java_11, won't it break something?

File: scripts/init.sh
Comment (graszka22) [inline scripts/init.sh:3]:
```suggestion
brew install swift-format ktlint release-it
```

File: .github/workflows/react_native_release.yaml
Comment (karkakol) [reply .github/workflows/react_native_release.yaml:2]: With jitpack all is necessary is tag. This workflow creates new tag so when react native is released then new tag is available for andoid jitpack. One thing that should be done is to somewhere in docks update to new tag.

File: compile_proto.sh
Comment (karkakol) [reply compile_proto.sh:13]: Why it is no longer required?

File: jitpack.yml
Comment (mironiasty) [inline jitpack.yml:1]: Ahh, so this is a trick to tell jitpack how to find android sdk in repo 😮

File: examples/video-chat/android/settings.gradle
Comment (karkakol) [reply examples/video-chat/android/settings.gradle:8]: it is not the case.
1. Jitpack versoin would be under adress of `com.github.organisation.repo.(tag | commit | branch)
2. Imports in kotlin files files are determined by `android-client-sdk/FishjamClient/build.gradle` via namespace
3. `:android-client' is only use in example app, so it will not brak anything

File: .github/workflows/android_build_lint_test.yaml
Comment (karkakol) [reply .github/workflows/android_build_lint_test.yaml:1]: This is quite fast task compared eg to react-native-e2e test so conditionally inwoke will not make anything faster.

File: packages/android-client/README.md
Comment (mironiasty) [inline packages/android-client/README.md:62]: To chyba można póki co wyrzucić

File: .github/workflows/android_build_lint_test.yaml
Comment (karkakol) [reply .github/workflows/android_build_lint_test.yaml:1]: But it can be done

File: packages/android-client/app/build.gradle
Comment (karkakol) [reply packages/android-client/app/build.gradle:1]: agreed, @mironiasty give us your opinion

File: scripts/init.sh
Comment (mironiasty) [inline scripts/init.sh:3]: Do we need `release-it` now?

File: packages/android-client/.release-it.json
Comment (karkakol) [reply packages/android-client/.release-it.json:1]: another thing to discuss

File: packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt
Comment (mironiasty) [inline packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt:62]: Formatting fixes in rn-client could be done in separate PR, it would help with understanding scope of changes in this one...

File: compile_proto.sh
Comment (graszka22) [reply compile_proto.sh:13]: it was required when rn client handled websockets communication
this code generates this stuff: https://github.com/fishjam-cloud/react-native-client-sdk/blob/3606a391982ed0314f084a30edfe0977b14a307b/src/protos/fishjam/peer_notifications.ts which was used for example here: https://github.com/fishjam-cloud/react-native-client-sdk/blob/3606a391982ed0314f084a30edfe0977b14a307b/src/FishjamClientContext.tsx#L103
I got rid of it recently and now the native sdks handle it

File: packages/react-native-client/android/build.gradle
Comment (karkakol) [reply packages/react-native-client/android/build.gradle:85]: it probably can be automated, but firstly we should discuss how we should think of process for deploying all packages (rn, android, ios)

Comment (mironiasty) [reply packages/react-native-client/android/build.gradle:85]: I hope we will be able to automate releasing all three SDKs once iOS one will be merged to this repo too...

File: packages/android-client/FishjamClient/build.gradle
Comment (karkakol) [reply packages/android-client/FishjamClient/build.gradle:16]: i copied it from code you gave me 😄 
I can check if it can be deleted.

File: examples/video-chat/android/settings.gradle
Comment (graszka22) [reply examples/video-chat/android/settings.gradle:8]: are you sure? when user uses our library in their app, it's autolinked similar to our example app. That means in Android Studio there will be a project named "fishjam-dev-react-native-client" which will have a build.gradle with those lines:
```
  if (findProject(':android-client')) {
    implementation project(':android-client')
```
This gradle file will be run when user builds his app. Now if in the user's app there is a project named "android-client" (I know it's really unlikely, but it's not nice to use such a generic name), then it'll be included as a dependency here instead of a remote library.

File: jitpack.yml
Comment (karkakol) [reply jitpack.yml:1]: 🪄

File: scripts/init.sh
Comment (karkakol) [reply scripts/init.sh:3]: For now i would keep it, then in another PR create whole configuration for releasing all packges

File: packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt
Comment (karkakol) [reply packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt:62]: For me it would be easier to leave it in this PR. If you **strongly** have another opinion i can rollback those changes and remove it from lint workflow.

File: examples/video-chat/android/settings.gradle
Comment (karkakol) [reply examples/video-chat/android/settings.gradle:8]: Ahh, I see your point

Comment (karkakol) [reply examples/video-chat/android/settings.gradle:8]: my proposition is then `fishjam-cloud-android-client`

Comment (graszka22) [reply examples/video-chat/android/settings.gradle:8]: ok 👍

File: packages/android-client/FishjamClient/build.gradle
Comment (karkakol) [reply packages/android-client/FishjamClient/build.gradle:16]: i unifided variables and removed not neccesary ones

File: compile_proto.sh
Comment (karkakol) [reply compile_proto.sh]: removed

File: packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt
Comment (karkakol) [reply packages/react-native-client/android/src/main/java/org/membraneframework/reactnative/VideoRendererView.kt:62]: removed

**Review by graszka22 — APPROVED** [review]
ok, but add tasks on linear about 
- removing android app
- adding linters
- adding release process

File: packages/android-client/FishjamClient/module.md
Comment (mironiasty) [inline packages/android-client/FishjamClient/module.md:3]: Great document! 👏

**Review by mironiasty — APPROVED** [review]
👍

**Pull request merged**
